module-system/sov-modules-macros/src/manifest.rs

```diff
@@ -188,42 +189,29 @@ impl Manifest {
                     v.as_u64().ok_or_else(|| {
                         Self::err(
                             &self.path,
-                            parent,
+                            &self.parent,
                             format!("`{}` attribute is not an array of integers", k),
                         )
                     })
                 })
                 .collect::<Result<Vec<_>, _>>()?;
 
-            let n = v.len();
-            fields.push(quote::quote!(pub #k: [u64; #n]));
-            field_values.push(quote::quote!(#k: [#(#v,)*]));
+            field_values.push(quote::quote!(#k: <<<Self as ::sov_modules_api::Module>::Context as ::sov_modules_api::Context>::GasUnit as ::sov_modules_api::GasUnit>::from_arbitrary_dimensions(&[#(#v,)*])));
         }
 
-        let ty = format!("{parent}GasConfig");
-        let ty = syn::parse_str(&ty).map_err(|e| {
-            Self::err(
-                &self.path,
-                parent,
-                format!("failed to parse type name `{}`: {}", ty, e),
-            )
-        })?;
-
-        let def = quote::quote! {
-            #[allow(missing_docs)]
-            #[derive(Debug, Default, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
-            pub struct #ty {
-                #(#fields,)*
-            }
-        };
+        // remove generics, if any
```

Comment:
Rust will not allow us to declare the type if it is defined via generics (i.e. reduce `Bank<C>` to `Bank`)